core/src/main/java/me/mykindos/betterpvp/core/effects/listeners/effects/FrozenListener.java

```diff
@@ -0,0 +1,97 @@
+package me.mykindos.betterpvp.core.effects.listeners.effects;
+
+import me.mykindos.betterpvp.core.effects.EffectManager;
+import me.mykindos.betterpvp.core.effects.EffectTypes;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.event.player.PlayerAttemptPickupItemEvent;
+import org.bukkit.event.player.PlayerDropItemEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.event.player.PlayerMoveEvent;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@BPvPListener
+@Singleton
+public class FrozenListener implements Listener {
+
+    private final EffectManager effectManager;
+
+    @Inject
+    public FrozenListener(EffectManager effectManager) {
+        this.effectManager = effectManager;
+    }
+
+    @EventHandler
+    public void onMove(PlayerMoveEvent event) {
+        if (event.hasChangedPosition()) {
+            if (effectManager.hasEffect(event.getPlayer(), EffectTypes.FROZEN)) {
+                event.setCancelled(true);
+            }
+        }
+    }
+
+    @EventHandler
+    public void onDropItem(PlayerDropItemEvent event) {
+        if (event.isCancelled()) return;
+        if (effectManager.hasEffect(event.getPlayer(), EffectTypes.FROZEN)) {
+            event.setCancelled(true);
+        }
+    }
+
+    @EventHandler
+    public void onPickupItem(PlayerAttemptPickupItemEvent event) {
+        if (event.isCancelled()) return;
+        if (effectManager.hasEffect(event.getPlayer(), EffectTypes.FROZEN)) {
+            event.setCancelled(true);
+        }
+    }
+
+    @EventHandler(priority = EventPriority.LOW)
+    public void onInteractEvent(PlayerInteractEvent event) {
```

Comment:
Might want to check if this is already cancelled, and the hand the player interacted with. Might end up sending multiple messages otherwise

Reply:
PlayerInteractEvent#isCancelled is deprecated, but I did change it to return if both of PlayerInteractEvent#useInteractedBlock and PlayerInteractEvent#useItemInHand are Event.Result.DENY (the full thing is cancelled)